docs/data/sdks/next-gen-sdks.md

```diff
@@ -0,0 +1,186 @@
+
+You can configure the storage provider in the Configuration of your Client. Refer to the "Configuration Options" section of a specific next-gen SDK full documentation for more details.
+
+### Logger provider
+
+A logger provider configure the logger instance used by Amplitude Client. It helps to collect debugging and error messages from the SDK in both development and production environments.
+
+You can configure the logger provider in the Configuration of your Client. Refer to the "Configuration Options" section of a specific next-gen SDK full documentation for more details.
+
+### Event 
+
+Events represent how users interact with your application. 
+
+`BaseEvent` represents a basic event with optional properties. You can also track other event properties in the field of `EventProperties`
+
+```python
+# Next-gen Python SDK example: Event
+
+from amplitude import Amplitude, BaseEvent
+
+client = Amplitude('API_KEY')
+
+# Track a basic event
+# One of user_id and device_id is required
+event = BaseEvent(event_type="Button Clicked", user_id="USER_ID")
+client.track(event)
+
+# Track events with optional properties
+client.track(
+    BaseEvent(
+        event_type="Button Clicked",
+        user_id="USER_ID",
+        app_version="1.2.1" # optional, the current version of your application
+        event_properties={ # other event properties you want to track 
+            "color": "red"
+        }
+))
+```
+
+### Plugin
+
+Plugins allow you to extend Amplitude SDK's behavior by, for example:
+
+- Use an Enrichment Plugin to modify event properties
+- Use a Destination Plugin to send events to a third-party APIs
+
+A plugin is an object with methods `setup()` and `execute()`:
+
+- `setup()` method contains logic for preparing the plugin for use and has Client instance as a parameter. The expected return value is None. A typical use for this method is to copy configuration from `client.configuration` or instantiate plugin dependencies. This method is called when you register the plugin to the client via `client.add()`.
+
+- `execute()` method the logic for processing events and has event instance as parameter. The expected return value for an Enrichment Plugin is the modified/enriched event, while the expected return value for a Destination Plugin is a map with keys: event (BaseEvent), code (HTTP response status code), and message (string). This method is called for each event that's instrumented using the client interface.
+
+## Common methods
+
+Next-gen SDKs share the same architecture as well as the same interfaces. Aligned interfaces benefit using Amplitude SDKs on multiple platforms without memorizing different function names.
+
+This a list of shared interfaces of next-gen SDKs. 
+
+!!!note 
+    Different SDKs follow different naming conventions. Here use camel case for methods and parameters and use pascal case for classes. 
+
```

Comment:
+ init ?

Reply:
Added `init`

Discussed in team sync and we agreed that event though some SDKs, like Python, don't have `init`, we still need to list it here because it is part of the common interfaces of next-gen SDKs.